src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/RazorHoverProvider.ts

```diff
@@ -0,0 +1,79 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as vscode from 'vscode';
+import { RazorLanguageFeatureBase } from './RazorLanguageFeatureBase';
+
+export class RazorHoverProvider
+    extends RazorLanguageFeatureBase
+    implements vscode.HoverProvider {
+
+    public async provideHover(
+        document: vscode.TextDocument, position: vscode.Position,
+        token: vscode.CancellationToken) {
+
+        const projection = await this.getProjection(document, position, token);
+        if (!projection) {
+            return;
+        }
+
+        const results = await vscode.commands.executeCommand<vscode.Hover[]>(
+            'vscode.executeHoverProvider',
+            projection.uri,
+            projection.position);
+
+        if (!results || results.length === 0) {
+            return;
+        }
+
+        // At the vscode.HoverProvider layer we can only return a single hover result. Because of this limitation we need to
+        // be smart about combining multiple hovers content or only take a single hover result. For now we'll only take one
+        // of them and then based on user feedback we can change this approach in the future.
+        const applicableHover = results.filter(item => item.range)[0];
+        if (!applicableHover) {
+            // No hovers available with a range.
+            return;
+        }
+
+        // Re-map the projected hover range to the host document range
+        const remappedResponse = await this.serviceClient.mapToDocumentRange(
+            projection.languageKind,
+            applicableHover.range!,
+            document.uri);
+
+        if (!remappedResponse) {
+            // Couldn't remap the projected hover location, there's no hover information available.
+            return;
+        }
+
+        if (document.version !== remappedResponse.hostDocumentVersion) {
```

Comment:
Is this something we need to be concerned about anywhere we call `mapToDocumentRange`? If so we probably need PR's for the work I did earlier. If not, how can I tell the difference?

Reply:
Ya we should. Probably makes sense to add the fix to your PRs as well